maskinporten_api/jwt_client.py

```diff
@@ -0,0 +1,133 @@
+"""Module for obtaining access tokens for Maskinporten.
+
+The main entry point is the `JWTAuthClient.get_access_token` method. Example
+usage:
+
+    conf = JWTConfig(
+        issuer=my_issuer,
+        consumer_org=my_consumer_org,
+        certificate=my_certificate,
+        private_key=my_private_key,
+    )
+    client = JWTAuthClient(conf, my_well_known_endpoint)
+    access_token = client.get_access_token(["list", "of", "scopes"])
+"""
+
+import base64
+import json
+import logging
+import uuid
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+
+import jwt
+import requests
+from OpenSSL import crypto
+
+
+class JWTAuthError(Exception):
+    pass
+
+
+@dataclass
+class JWTConfig:
+    issuer: str
+    consumer_org: str
+    certificate: crypto.X509
+    private_key: crypto.PKey
+
+
+@dataclass
+class JWTGenerator:
+    jwt_config: JWTConfig
+
+    def _jws_headers(self):
+        """Return JWT headers for the present certificate."""
+        x5c = base64.b64encode(
+            crypto.dump_certificate(crypto.FILETYPE_ASN1, self.jwt_config.certificate)
+        )
+        return {
+            "x5c": [x5c.decode("utf-8")],
+            "alg": "RS256",
+        }
+
+    def _claims(self, audience, scopes):
+        """Return a JWT payload (a set of claims) for `audience` and `scopes`.
+
+        The expiration time is currently hard-coded to 120 seconds. Both issue
+        time (iat) and expiration time (exp) are given as POSIX timestamps.
+        """
+        now = datetime.now()
+        expiration_time = now + timedelta(seconds=120)
+
+        return {
+            "aud": audience,
+            "exp": int(expiration_time.timestamp()),
+            "iat": int(now.timestamp()),
+            "iss": self.jwt_config.issuer,
+            "jti": str(uuid.uuid4()),  # Must be unique for each grant.
+            "scope": " ".join(scopes),
+        }
```

Comment:
Burde det også være noe for `consumer_org` her? https://github.com/oslokommune/maskinporten-cli/blob/main/src/main/kotlin/no/kommune/oslo/jwt/JwtGenerator.kt#L38-L40

Reply:
Yup! Den glemte jeg.